app/(protected)/card/activate.tsx

```diff
@@ -0,0 +1,415 @@
+            setKycStatus(customer.kycStatus);
+          } else {
+            console.log("Keeping kycStatus from URL params:", kycStatus);
+          }
+
+          const kycLinkData = await withRefreshToken(
+            getKycLink(customer.kycLinkId)
+          );
+
+          if (!kycLinkData) throw new Error("Error getting kyc link");
+
+          setKycLink(kycLinkData);
+        }
+      } catch (error) {
+        console.error("Error checking customer status:", error);
+      }
+    };
+
+    checkCustomerStatus();
+  }, []); // Keep it with no dependencies
+
+  const handleProceedToTos = async () => {
+    try {
+      let kycLinkData: KycLink;
+
+      if (kycLink) {
+        kycLinkData = kycLink;
+      } else {
+        const kycLinkOrUndefined = await withRefreshToken(
+          createKycLink(fullName, email, getRedirectUrl())
+        );
+
+        if (!kycLinkOrUndefined) {
+          throw new Error("Error creating kyc link");
+        }
+
+        kycLinkData = kycLinkOrUndefined;
+        setKycLink(kycLinkData);
+      }
+
+      setIsLoading(true);
+
+      router.push({
+        pathname: path.CARD_TERMS_OF_SERVICE,
+        params: {
+          url: kycLinkData.tosLink,
+        },
+      });
+    } catch (error) {
+      console.error("Error proceeding to ToS:", error);
+    } finally {
+      setIsLoading(false);
+    }
+  };
+
+  const handleProceedToKyc = async () => {
+    try {
+      let kycLinkData: KycLink;
+
+      if (kycLink) {
+        kycLinkData = kycLink;
+      } else {
+        const redirectUrl = getRedirectUrl();
+
+        const kycLinkOrUndefined = await withRefreshToken(
+          createKycLink(fullName, email, redirectUrl)
+        );
+
+        if (!kycLinkOrUndefined) {
+          throw new Error("Error creating kyc link");
+        }
+
+        kycLinkData = kycLinkOrUndefined;
+        setKycLink(kycLinkData);
+      }
+
+      setIsLoading(true);
+
+      router.push({
+        pathname: path.CARD_KYC,
+        params: {
+          url: kycLinkData.link,
+        },
+      });
+    } catch (error) {
+      console.error("Error proceeding to KYC:", error);
+    } finally {
+      setIsLoading(false);
+    }
+  };
+
+  const getRedirectUrl = () => {
+    const baseUrl = process.env.EXPO_PUBLIC_BASE_URL;
+    return `${baseUrl}${path.CARD_ACTIVATE}?kycStatus=${KycStatus.UNDER_REVIEW}`;
+  };
+
+  const handleActivateCard = async () => {
+    try {
+      setIsLoading(true);
+      // Add your card activation logic here
+      console.log("Activating card...");
+      // For demo purposes - in real implementation this would be an API call
+      setTimeout(() => {
+        setCardActivated(true);
+        setIsLoading(false);
+      }, 1000);
+      // Navigate to successful activation or card details screen
+    } catch (error) {
+      console.error("Error activating card:", error);
+      setIsLoading(false);
+    }
+  };
+
+  const handleRefreshKyc = async () => {
```

Comment:
@SametSahin10 Tanstack Query `refetch` and `isRefetching` can be used instead of handleRefreshKyc and isRefreshing:
https://tanstack.com/query/latest/docs/framework/react/reference/useQuery